src/FileStorageBase/Cache/ClientCache/MultiClient.cs

```diff
@@ -0,0 +1,604 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Neo.FileStorage.API.Accounting;
+using Neo.FileStorage.API.Acl;
+using Neo.FileStorage.API.Container;
+using Neo.FileStorage.API.Netmap;
+using Neo.FileStorage.API.Object;
+using Neo.FileStorage.API.Refs;
+using Neo.FileStorage.API.Reputation;
+using Neo.FileStorage.API.Session;
+using Neo.FileStorage.API.Client;
+using UsedSpaceAnnouncement = Neo.FileStorage.API.Container.AnnounceUsedSpaceRequest.Types.Body.Types.Announcement;
+using System.Collections.Concurrent;
+
+namespace Neo.FileStorage.Cache
+{
+    public class MultiClient : IFSClient, IFSRawClient
+    {
+        private readonly List<Network.Address> addresses;
+        private readonly ConcurrentDictionary<Network.Address, Client> clients = new();
+
+        public MultiClient(List<Network.Address> addrs)
+        {
+            addresses = addrs;
+        }
+
+        public void Dispose()
+        {
+            lock (clients)
+            {
+                addresses.Clear();
+                foreach (var cli in clients.Values)
+                    cli.Dispose();
+                clients.Clear();
+            }
+        }
+
+        public IFSRawClient Raw()
+        {
+            return this;
+        }
+
+        private Client Client(Network.Address address)
+        {
+            if (!clients.TryGetValue(address, out var client))
+            {
+                client = new(null, "http://" + address.ToHostAddressString());
```

Comment:
Why not `https`? add extra argument?

Reply:
`http` is tentative. Will support `https` later.

Reply:
Updated.